File: solutions/c/square-root/1/square_root.c
#include "square_root.h"

int square_root(int n){
    int a = (n/2 > 0) ? n/2 : 1, b = 0;
    while (a != b){
        b = a;
        a = (a + n/a)/2;
    }
    return a;
}

File: solutions/julia/square-root/1/square-root.jl
function square_root(n)
    if n < 0
        throw(DomainError(n))
    end
    a = n÷2 > 0 ? n÷2 : 1
    b = 0
    while a != b
        b = a
        a = (a + n÷a) ÷ 2
    end
    a
end